### PR TITLE
Feature: Display self-change blockers

### PR DESCRIPTION
**Pull request by maxplt**
# Jira Issue: [IP-211](https://hedvig.atlassian.net/jira/software/projects/IP/boards/19?selectedIssue=IP-211)

## What?
Show self-change blockers on a member's contract tab if there exists

Not sure which values can be, so I've added colors for `termination` and `future changes` types. And `accent` color by **default**. 

## Screenshots
Снимок экрана 2021-12-13 в 17 28 45
Снимок экрана 2021-12-13 в 17 29 27

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [ ] Codescouted
- [ ] Unit tests written
- [ ] Tested locally

### CODE REVIEW

File: src/features/member/tabs/contracts-tab/contract/index.tsx [hunk 1]
```diff
@@ -21,12 +22,52 @@ import {
...
+}
+
+const getSelfBlockerNormalName = (value: string) => {
+  const lowercaseValue = value
+    .split('_')
+    .map((word) => word.toLowerCase())
+    .join(' ')
+
+  return lowercaseValue.charAt(0).toUpperCase() + lowercaseValue.slice(1)
+}
```
Comment (rasmusguterstam) [inline src/features/member/tabs/contracts-tab/contract/index.tsx:37]: Use existing utils instead, something like `convertEnumToSentence`

**Review by rasmusguterstam — APPROVED** [review]
Really cool! Maybe have a sync with Hunt on the layout? I'm thinking that maybe it's not something that should always be top-level information for IEX, since self-change is just one feature 🤔 

I'll approve, but check with him first 😄

**Comment by maxplt** [comment]
## New view

Снимок экрана 2021-12-15 в 10 25 42
Снимок экрана 2021-12-15 в 10 25 52

**Comment by rasmusguterstam** [comment]
> ## New view
> Снимок экрана 2021-12-15 в 10 25 42
> 
> Снимок экрана 2021-12-15 в 10 25 52

Yeah this ain't it. The blockers are not the action in itself, they are the reasons as to why the user can't do self-change (which currently is to change address, this was called "Moving flow" before)

File: src/features/member/tabs/contracts-tab/contract/index.tsx [hunk 2]
```diff
@@ -17,16 +18,44 @@ import {
   FocusItems,
   useNavigation,
 } from 'features/navigation/hooks/use-navigation'
-import React, { useRef } from 'react'
+import React, { useRef, useState } from 'react'
+import { ExclamationCircle } from 'react-bootstrap-icons'
 import { Contract as ContractType } from 'types/generated/graphql'
 import { useElementFocus } from '@hedvig-ui/hooks/use-element-focus'
 
+const blockersTranslates = {
```
Comment (rasmusguterstam) [inline src/features/member/tabs/contracts-tab/contract/index.tsx:26]:
```suggestion
const blockerDisplayName = {
```

File: src/features/member/tabs/contracts-tab/contract/index.tsx [hunk 3]
```diff
@@ -17,16 +18,44 @@ import {
...
 
+const Blockers = styled.div`
+  margin: 0;
+  padding: 0;
+  list-style: none;
+
+  display: flex;
+  align-items: center;
+
+  background-color: ${({ theme }) => theme.accentLighter ?? colorsV3.white};
```
Comment (rasmusguterstam) [inline src/features/member/tabs/contracts-tab/contract/index.tsx:45]: What is this default behavior about?

File: shared/@hedvig-ui/OrbIndicator/orb-indicator.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ export const OrbIndicator: React.FC<OrbIndicatorProps> = ({
       : lightTheme.placeholderColor
 
   return (
-    <OrbIndicatorStyled color={currentColor} size={size || 'tiny'} {...props} />
+    <OrbIndicatorStyled color={currentColor} size={size || '10px'} {...props} />
```
Comment (rasmusguterstam) [inline shared/@hedvig-ui/OrbIndicator/orb-indicator.tsx:34]: Why not style it inline if you really need 10px in one place?

**Review by rasmusguterstam — CHANGES_REQUESTED** [review]
I think there's some misunderstanding as to what these blockers mean 😅

File: src/features/member/tabs/contracts-tab/contract/index.tsx (hunk 3 above)
Comment (maxplt) [reply src/features/member/tabs/contracts-tab/contract/index.tsx:45]: I just copied that kind of _background_ from `Card` _styles_ 🥲

File: shared/@hedvig-ui/OrbIndicator/orb-indicator.tsx (hunk 4 above)
Comment (maxplt) [reply shared/@hedvig-ui/OrbIndicator/orb-indicator.tsx:34]: Sorry, I didn't understand what you mean 😓
Do you mean to use a construction like this?
`<OrbIndicatorStyled style={{ height: size || '10px', width:  size || '10px' }} {...props} />`

**Comment by hunt-hedvig** [comment]
Sorry for the hassle, I simply relocate the blocker to the bottom of each contract. let's see if this will fit ;)
@rasmusguterstam @maxplt 

image

**Comment by maxplt** [comment]
## New view again!

Снимок экрана 2021-12-16 в 16 40 03

File: shared/@hedvig-ui/OrbIndicator/orb-indicator.tsx (hunk 4 above)
Comment (rasmusguterstam) [reply shared/@hedvig-ui/OrbIndicator/orb-indicator.tsx:34]: No I just mean why default to 10px, if you need 10px in once place then just use the inline style for it in that particular instance

**Review by rasmusguterstam — APPROVED** [review]
Nice, now we're on the same page 😄

**Pull request merged**